### ISSUE TITLE
What is the ISO27001 view on password expiry ?

### ISSUE THREAD

**Issue by Melonbwead** [dwyl/ISO-27001-2013-information-technology-security#24]
I know this use to be enforced in the earlier version of ISO-27001, but i can't find anything on password expiry in here, maybe i haven't looked hard enought

**Comment by nelsonic** [comment]
Hi @ChennyBaBy, good question! (_thank you for opening this issue!_)

Sadly, the ISO27001-2013 doc https://trofisecurity.com/assets/img/iso27001-2013.pdf
Has 3 instances of the word password and they are all in the same "control" (A.9.4.3):
![image](https://user-images.githubusercontent.com/194400/35330305-fdec4dce-00fa-11e8-9b95-28a203b85ba4.png)

> "_**Password management systems** shall be **interactive** and shall **ensure quality passwords**._"

It does _not_ mention passwords _must_ be rotated ... 🙄 

## tl;dr: ISO27001-2013 Does _Not_ Require Password Rotation

Typically most "Enterprise" (Corporate) environments require monthly password changes.
But, as **Bruce Schneier** (_who we **highly respect**_) says in his **10<sup>th</sup>October 2017** article "Changes in Password Best Practices": https://www.schneier.com/blog/archives/2017/10/changes_in_pass.html
![image](https://user-images.githubusercontent.com/194400/35330571-e73ef878-00fb-11e8-8bd9-b196d7cb7873.png)

> 2. "_Stop it with password expiration. That was an old idea for an old way we used computers. Today, don't make people change their passwords unless there's indication of compromise._"

From ***personal experience*** (_working in "high security" environments_) passwords are generally a "weak link", if "complexity" is required people either easily forget their password, or they use a predictable pattern like "word" + digits + punctuation. which when they are required to change their password, they simply increment the digits e.g: "**Unicorns7$**" >> "**Unicorns8$**" this satisfies the criteria/requirements: 
+ Length: at least 10 characters.
+ Mixed case: at least 1 upper case character.
+ Digits: at least 1 digit.
+ Special characters: at least 1 special character.

As you can tell this is a _very_ predictable pattern and is not especially "_secure_"...

A _much_ better way of encouraging/enforcing security is:
+ Educate users to use a "***passphrase***" not a "**password**" 
i.e. use ***several*** words that make up a phrase that is meaningful to them and they can easily remember: e.g:
  + "**Bigbenisthebell1859!**" (20 chars) a good passphrase for a "history buff" https://en.wikipedia.org/wiki/Big_Ben 
  + "**ChewbaccaisaWookiee1977.**" (24 chars) for someone keen on StarWars https://en.wikipedia.org/wiki/Wookiee
+ Limit the number of "attempts" a person type an incorrect passphrase e.g: 10 tries.
+ Remind people to **_never_ write down or share** their passphrase.
+ Make it fast and easy for people to _reset_ their password if they forget it.
+ Raise upper-limit on password length to **64 Chars** to accommodate for Strong Passwords generated by a Password Manager like "Last Pass" or "1Password". see: https://www.grc.com/passwords.htm and 

![xkcd-password-strength](https://user-images.githubusercontent.com/194400/35330930-4fd53c48-00fd-11e8-969a-a2824109c103.png)
Source: https://xkcd.com/936

If you're _curious_ I would recommend reading the discussion/answers on this thread:
https://security.stackexchange.com/questions/4704/how-does-changing-your-password-every-90-days-increase-security
This is also a _good_ article "**Time to rethink mandatory password changes**": https://www.ftc.gov/news-events/blogs/techftc/2016/03/time-rethink-mandatory-password-changes

**Comment by Melonbwead** [comment]
Thank you so much for this amazing detailed reply!!!! ❤️

**Comment by Truck3r** [comment]
However ISO 27002 _does_ specify rules for password expiry and rotation:
Section 11.3.1 e)
_change passwords at regular intervals or based on the number of accesses (passwords for
privileged accounts should be changed more frequently than normal passwords), and
avoid re-using or cycling old passwords;_ 

Since 27002 is often used as implementation guidelines for 27001 (as I understand it), you will likely run into password expiry and rotation in organisations that are 27001 certified. Unfortunately 😢

**Comment by dyfrgi** [comment]
The language about changing passwords was in ISO 27002:2005, but does not appear in either the 2013 or 2022 standards. The section which previously said that is now 5.17, which you can read at https://itref.ir/uploads/editor/d3d149.pdf. A note at the end of that section says this:

> Requiring frequent change of passwords can be problematic because users can get annoyed by the
frequent changes, forget new passwords, note them down in unsafe places, or choose unsafe passwords.
